jbake-core/src/main/java/org/jbake/app/Oven.java

```diff
@@ -154,38 +154,39 @@ public void bake() {
         setLocale();
 
         try {
-
-            final long start = new Date().getTime();
-            LOGGER.info("Baking has started...");
             contentStore.startup();
             updateDocTypesFromConfiguration();
             contentStore.updateSchema();
             contentStore.updateAndClearCacheIfNeeded(config.getClearCache(), config.getTemplateFolder());
 
+            LOGGER.info("Baking has started...");
+            final long bakeStart = new Date().getTime();
             // process source content
+            LOGGER.info("Crawling content...");
             crawler.crawl();
 
-            // process data files
-            crawler.crawlDataFiles();
-
             // render content
+            LOGGER.info("Rendering content...");
             renderContent();
 
             // copy assets
+            LOGGER.info("Copy assets...");
             asset.copy();
             asset.copyAssetsFromContent(config.getContentFolder());
 
+            utensils.getRenderer().shutdown();
             errors.addAll(asset.getErrors());
```

Comment:
shutdown() does not wait for the ES to run all threads until finished (ie non-blocking operation). You need for all threads to finish, or you will not get all asset errors.